### PR TITLE
[13.0][FIX] l10n_es: Includes new food taxes reduction

### PR DESCRIPTION
**Pull request by pedrobaeza**
Supersede of #108846

BOE-A-2022-22685 Real Decreto-ley 20/2022, de 27 de diciembre: https://www.boe.es/buscar/doc.php?id=BOE-A-2022-22685

Changes over previous PR:

- Fix duplicated XML-IDs.
- Rename taxes to explicitly mention "Alimentos", for avoiding confusions on product taxes selection. Anyway, the reusing of these taxes in the future is uncertain, and if any, it would be debatable to have different concepts in the same tax over time.
- Include "recargo de equivalencia" taxes.
- Include intra and extra community taxes.
- Include taxes mappings for the last 2.

@Tecnativa TT40976

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Ping @jco-odoo @oco-odoo @dylankiss

**Comment by rvalyi** [comment]
(authored by "Pedro the thief" ;-D )
just a joke to tell you you have all our support from Akretion.

**Review by rafaelbn — APPROVED** [review]
👍 Thank you!

**Comment by acysos** [comment]
Hello,

I understand that this is a bug in the id of one the taxes and I the "Recargo Equiovalencia" is neccesarry. I think that "Inctracomunitario" and "Extracomunitario" is not neccesary, it's very extrange this case, but I understand that is better add it.

But this a problem. You supposed that is only for food, but we don't know if the future the spanish goverment user this tax for other product. But now this code in production because today is mandatory. If other prodcuts will be added you can fix in the future.

Greetings

**Comment by pedrobaeza** [comment]
The intra and extra community extra taxes and fiscal positions is due to this:

Selección_001

we don't know if they are going to be declared in a separate field. At least, they will be joined in the corresponding intra/extra fields, so we need such extra taxes.

About the name, I have already written this: `Rename taxes to explicitly mention "Alimentos", for avoiding confusions on product taxes selection. Anyway, the reusing of these taxes in the future is uncertain, and if any, it would be debatable to have different concepts in the same tax over time.`

When we have another case for such tax percentage, we can:

- Create a new one for not mixing.
- Rename this one to be more generic.

But meanwhile, we don't confuse people about which tax to select between the 0% / exempt / not subjected ones.

**Comment by jco-odoo** [comment]
@rafaelbn @pedrobaeza Do not we need a link with the tags of the modelo 303 as well?

**Comment by pedrobaeza** [comment]
The problem is that right now, AEAT hasn't published yet the way it's going to be informed in the 303 model. Imagine how this has going: announced by the government on 27th of December, published in BOE on 28th of December, and to start with on 1st of January... Crazy.

I think it's better to not put tags for now and then add them when known.

**Comment by jco-odoo** [comment]
@chklop

**Comment by jco-odoo** [comment]
@pedrobaeza Then we can expect a new version of the 303 because in https://sede.agenciatributaria.gob.es/static_files/Sede/Disenyo_registro/DR_300_399/archivos_23/DR303e23.xlsx I don't see e.g. Recargo Equivalencia de 0.625%

**Comment by jco-odoo** [comment]
@pedrobaeza @rafaelbn We would wait for another week then hoping the AEAT will shine a little bit more light on this and otherwise we can still merge like this (or maybe add a 'preliminary tag' or something).  From v14, we could also add things to automatically update the CoA in existing companies like for l10n_lu.

**Comment by pedrobaeza** [comment]
Let's wait then for the definitive 303 changes if you prefer.

**Comment by rafaelbn** [comment]
Hello @jco-odoo ,

My opinion is merge this today. Companies should account invoices today, but modelo 303 of invoices of January need to be presented 20th February.

So we can merge this and we have month and a  half yo update tags

Thank you! 😄 

m2c

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

Code review and test

**Review by acysos — APPROVED** [review]

**Comment by chklop** [comment]
Adding tags afterwards is kind of an annoyance, especially if transactions using these taxes have already been made. What about us setting up a temporary tax tag for those new takes (something like 303_food_5) that we will pick up and put wherever it will be decided by AEAT in the next version of the 303 ?

**Comment by acysos** [comment]
Hello,

I understand that can be a annoyance, but we can do nothing against a law and goverment decision. If you think that is neccesary put the 303 tag now. Put a temporaty tag can be a solution.

Greetings

**Comment by jco-odoo** [comment]
@acysos Could you think about some preliminary tags we can provide?  (like only base or only tax, will the recargo be by rate or just one for all?)

**Comment by pedrobaeza** [comment]
@chklop If we put such temporary tags, we will base and fee tags for each tax, which are 24 new tags. Do you really want that?

**Comment by pedrobaeza** [comment]
A change from AEAT: Recargo de equivalencia 0.625% converts to 0.62%, as they won't adapt all their infrastructure to support 3 decimals in the tax percentage. They have also change the criteria of the 0% tax from exempt to subjected (but doesn't affect here).

Still no news on the 303 model.

**Comment by jco-odoo** [comment]
@pedrobaeza @acysos Looks like we will need an extra tag for casillas 156 and 158. https://www.acysos.com/blog/noticias-1/ultima-hora-modelo-303-y-sii-3

**Comment by jco-odoo** [comment]
@chklop

**Comment by chklop** [comment]
Good catch @jco-odoo.
Tax tag 156 is already added via this PR which is intended to be merged very soon (Modelo 303 version 2023, already tested functionally https://github.com/odoo/odoo/pull/103361/files). Once it's done, this PR could reuse the tag then.

**Comment by pedrobaeza** [comment]
That won't be the definitive changes though.

**Review by acysos — APPROVED** [review]

**Comment by acysos** [comment]
Hello,

The 303 structure is published: https://sede.agenciatributaria.gob.es/static_files/Sede/Disenyo_registro/DR_300_399/archivos_23/DR303e23.xlsx

Greetings

**Comment by pedrobaeza** [comment]
Can you summarize the needed tags then?

**Comment by dylankiss** [comment]
> The 303 structure is published: https://sede.agenciatributaria.gob.es/static_files/Sede/Disenyo_registro/DR_300_399/archivos_23/DR303e23.xlsx

Any idea why there are 3 values mentioned for casilla 17: (00000) 0%, (00050) 0.5% and (00062) 0.62%?
What needs to go in there and when?

**Comment by jco-odoo** [comment]
@dylankiss I suppose these are the recargo de equivalencia taxes.

**Comment by dylankiss** [comment]
@jco-odoo Yes, I suppose too, but what value should we mention in the BOE export? It seems not specified in the file.

**Comment by jco-odoo** [comment]
@dylankiss It is about where to put [17]?  There is no updated spec of the BOE as well?  (maybe there is some reserved space that will be filled with that)

**Comment by chklop** [comment]
No reserved space it seems. I'd put any of the three value there tbh. Doesn't really make much sense, but this is clearly done in a hurry.

**Comment by jco-odoo** [comment]
@chklop We could divide [18] by [16] and the one that is the closest...

**Comment by chklop** [comment]
> @chklop We could divide [18] by [16] and the one that is the closest...

I don't think that's right. That space is normally foreseen to put a Constant in there. But they need to "shove" some of those new tax rates somewhere. Seems like AEAT decided it should go there, but then the "00000" constant doesn't really make much sense. My guess is they decided to authorize any of the proposed constant as valid. Probably temporarily, and a new version will be coming with a space for each one ?

**Comment by jco-odoo** [comment]
@chklop Why would not it make any sense?  If your recargo was always 0%, then you put 0, if it was always 0.62, you put the other thing...

**Comment by chklop** [comment]
> @chklop Why would not it make any sense? If your recargo was always 0%, then you put 0, if it was always 0.62, you put the other thing...

So you think it's impossible to have a combination of those tax rates applicable ?

**Comment by pedrobaeza** [comment]
I have checked the new BOE format and they continue doing some mess. Technically, you can have in the same company all three "recargos de equivalencia" 0%, 0.5% and 0.62%, as they depend on the product nature. On OCA, I'm going to implement it taking the first percentage found (if any, or 0 otherwise).

**Comment by jco-odoo** [comment]
@dylankiss

**Comment by jco-odoo** [comment]
@pedrobaeza @acysos Is it still possible to test the 303 online?  (for the 390 it seems possible)

Also, we know how to report the recargo now, but do we have more info about the food taxes.  Because, why do we split the taxes if they are not reported differently?

**Comment by pedrobaeza** [comment]
> Is it still possible to test the 303 online? (for the 390 it seems possible)

The tool I know is:

https://preportal.aeat.es/PRE-Exteriores/Inicio/_menu_/IVA/Modelo_303/Modelo_303.html

but it also requires an Spanish company certificate. Do you already have yours on the Valencia subsidiary?

> Also, we know how to report the recargo now, but do we have more info about the food taxes. Because, why do we split the taxes if they are not reported differently?

Although in model 303 may not have different fields for putting services and goods, we already have the experience that AEAT asks for differentiating intra-community services and goods, so the only way to map these intra-community taxes is to have also split the national ones. But that's not only the case where we need to split goods and services. There has been several occasions that in the model 390 (yearly summary of the 303), it has been required to declare isolatedly national goods and services.

**Comment by rafaelbn** [comment]
Hello! Are you going to wait until models are ready, right?

We need in v15 this new taxes OPW-3143902 

 @chklop @jco-odoo maybe you could open the PR for 14, 15, 16 so we can add the PR to our customers in order invoice customers and get bill from suppliers.

Let me know please

Thank you!

**Comment by jco-odoo** [comment]
@rafaelbn https://github.com/odoo/odoo/pull/103361#issuecomment-1398462093

**Comment by rafaelbn** [comment]
Thank you @jco-odoo !

**Comment by pedrobaeza** [comment]
So no merge on v13?

**Comment by jco-odoo** [comment]
@pedrobaeza Normally, 13 is not supported anymore, but if you want to do the backport or it does not give many problems to do so, we can still merge. Not sure if we have the code for updating the taxes however.

File: addons/l10n_es/data/account_tax_data.xml
Comment (rafaelbn) [inline addons/l10n_es/data/account_tax_data.xml:1999]: I guess the equivalent for 14, 15 is

https://github.com/odoo/odoo/blob/15.0/addons/l10n_es/data/account_tax_data.xml#L2575

```
    <record id="account_tax_template_s_iva0b" model="account.tax.template">
        <field name="type_tax_use">sale</field>
        <field name="name">IVA 0% (Bienes)</field>
        <field name="chart_template_id" ref="l10n_es.account_chart_template_common"/>
        <field name="amount" eval="0"/>
        <field name="amount_type">percent</field>
        <field name="tax_group_id" ref="tax_group_iva_0"/>
        <field name="invoice_repartition_line_ids" eval="[(5, 0, 0),
```

Comment (dylankiss) [reply addons/l10n_es/data/account_tax_data.xml:1999]: @rafaelbn Correct for the 0% food taxes, yes

Comment (pedrobaeza) [reply addons/l10n_es/data/account_tax_data.xml:1999]: Oof, you have changed the XML-IDs and names we put in this PR, and it was for a good reason. Now everything is getting complicated...

Comment (rafaelbn) [reply addons/l10n_es/data/account_tax_data.xml:1999]: Exactly, `account_tax_template_s_iva0_a` vs `account_tax_template_s_iva0b` and so on , that why my comment.

**Comment by acysos** [comment]
@pedrobaeza remember add the fiscal position REAGYP, that is commeted in the OCA l10n_es_extra_data for Odoo 12.0 https://github.com/OCA/l10n-spain/pull/2716

Greetings

**Comment by rafaelbn** [comment]
@pedrobaeza Odoo forgot to update REAGYP fiscal position, maybe you want to update here too with all mappings: https://github.com/odoo/odoo/pull/110592#issuecomment-1412634255

**Comment by rafaelbn** [comment]
We have an agriculture/distributor/b2c client and this is the use case:

1. They have to send something like a bill name "Recibo agrario", this document is in like for supplier. This is only for some suppliers not all, just the supplier which you set with the fiscal position "REAGYP"
2. So if the product is "**natural**" then product has set tax "0% IVA soportado (bienes corrientes)" for general suppliers that are not in the REAGYP
3. When you create the "bill" (Recibo agrario) then as supplier has REAGYP but there is not mapping taxes doesn't change and maintain "0% IVA soportado (bienes corrientes)" so they have to change manually for 12% + IRPF 2%

In our case is with "Pistachos"

@acysos said the same here: https://github.com/OCA/l10n-spain/pull/2716#issuecomment-1403391401

> Ok, los cambio y los subo. Pero hemos visto otro tema, que hay que hacerlo, no se ha modificado la posición fiscal de regimen agrario que si puede vender aceite o cereales que han cambiado. Vamos a preparar ambos y los subimos.

**Comment by pedrobaeza** [comment]
Hi @jco-odoo @dylankiss I have cherry-pick Dylan's commit from 14.0 removing the migration script part as this version doesn't have that part handled, but it's very interesting this to be merged for 13.0, as still a lot of people didn't have the occasion to migrate to upper versions. Can you please consider merging it?

@rafaelbn I see very weird that a 0% tax is converted to 12% on that regime. Anyway, it seems Odoo is handling it in #112023

**Comment by jco-odoo** [comment]
@fw-bot ignore

**Comment by jco-odoo** [comment]
@Levizar LGTM, but if you could check before merging, that would be fine.  

@pedrobaeza Could you confirm you tested it?

**Comment by pedrobaeza** [comment]
Yes, already tested in three 13.0 installations, and even migration scripts for OpenUpgrade has been created and tested: https://github.com/OCA/OpenUpgrade/pull/3716

**Comment by pedrobaeza** [comment]
@dylankiss @jco-odoo check commit 83977eb2c8ed43fb5004271b43ab75fab8e4cf20, which contains fixes for the PR merged in v14 (and included here in this PR), which should be forward-ported.

**Comment by Levizar** [comment]
Hello @pedrobaeza , nearly LGTM . 
Could you just reword @dylankiss 's commit to add this line at the end?
> backport of b4cd54a3c9227370667913ff2e170e24539c6328

If that's ok for you, @dylankiss will take care of cherry-picking your commit on the 14.0 and forwardporting it.

**Comment by pedrobaeza** [comment]
Done @Levizar

Thanks.

**Comment by Levizar** [comment]
@pedrobaeza It looks like something went wrong :rofl: 
image

**Comment by pedrobaeza** [comment]
Wow, indeed... Checking what's happening.

**Comment by Levizar** [comment]
Maybe you rebased on the branch of the OCA repository?

**Comment by pedrobaeza** [comment]
Ok, solved. My branch was based on OCA/OCB instead of odoo/odoo. Sorry for the slip.

**Comment by Levizar** [comment]
Shit happens :grin: 

Seems like the hypertext isn't working on the commit because it's not the full id of the commit.
Would you mind changing it to `b4cd54a3c9227370667913ff2e170e24539c6328` ?
If not, that's not a big deal.

**Comment by pedrobaeza** [comment]
Changed the commit message to full SHA. Not sure if still on time, as I don't know how runbot acts on this.

**Comment by Levizar** [comment]
Nice, thanks!
Regarding runbot, as soon as you force push, the PR lose it's approval.

**Comment by Levizar** [comment]
@robodoo r+ rebase-ff

**Comment by Levizar** [comment]
At any moment, if you are wondering what's the status of the current PR, you can click on the link that robodoo post at the beginning of the PR
image

You have a small recap of the PR status
image

**Comment by pedrobaeza** [comment]
Thanks for the extra info about runbot. Noted for other times.

**Comment by ljsalvatierra-factorlibre** [comment]
Hi @pedrobaeza 

I am currently reviewing these changes and have a question regarding `IVA 0% (Servicios)` (`account_tax_template_s_iva0s`) and `IVA 0% (Bienes)` (`account_tax_template_s_iva0b`).

The tags for this Tax Template are:

- `mod_303_153`
- `mod_303_14_sale`

But the report `mod303` needs:

- `mod_303_14_sale`
-  `mod_303_15_sale`
-  `mod_303_150`
-  `mod_303_152`

Could you please tell me if i am missing something?

**Comment by pedrobaeza** [comment]
For that ones, I didn't do any change from v14, but we don't use tax tags in OCA/l10n-spain, so don't care too much. The other were changed because it affects the repartition lines. If they are not correct, please do a PR amending them.

**Comment by ljsalvatierra-factorlibre** [comment]
> but we don't use tax tags in OCA/l10n-spain

Thank you for the clarification :)

**Pull request closed without merging**
